testing/src/test/java/org/oppia/android/testing/espresso/TextInputActionTest.kt

```diff
@@ -0,0 +1,110 @@
+package org.oppia.android.testing.espresso
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.android.material.textfield.TextInputLayout
+import com.google.common.truth.Truth.assertThat
+import org.hamcrest.Description
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.testing.R
+import org.oppia.android.testing.TextInputActionTestActivity
+import org.oppia.android.testing.espresso.TextInputAction.Companion.ErrorTextExisted
+import org.oppia.android.testing.espresso.TextInputAction.Companion.ErrorTextNotExisted
+import org.robolectric.annotation.LooperMode
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class TextInputActionTest {
+
+  @get:Rule
+  var activityRule =
+    ActivityScenarioRule<TextInputActionTestActivity>(
+      TextInputActionTestActivity.createIntent(ApplicationProvider.getApplicationContext())
+    )
+
+  lateinit var context: Context
+
+  @Before
+  fun setupContext() {
+    activityRule.scenario.onActivity {
+      context = it.baseContext
+    }
+    context.setTheme(R.style.Theme_AppCompat_Light)
+  }
+
+  @Test
+  fun testTextInputAction_ErrorTextExisted_matchesSafelyTrue() {
+    val expectedErrorText = "Incorrect Administrator PIN. Please try again."
+    val textInputLayout = TextInputLayout(context)
+
+    textInputLayout.error = "Incorrect Administrator PIN. Please try again."
+
+    val errorTextExisted = ErrorTextExisted(expectedErrorText)
+    val result: Boolean = errorTextExisted.matchesSafely(textInputLayout)
+    assertThat(result).isTrue()
+  }
+
+  @Test
+  fun testTextInputAction_ErrorTextExisted_matchesSafelyFalse() {
+    val expectedErrorText = "Incorrect Administrator PIN. Please try again."
+    val textInputLayout = TextInputLayout(context)
+
+    textInputLayout.error = "This name is already in use by another profile."
+
+    val errorTextExisted = ErrorTextExisted(expectedErrorText)
+    val result: Boolean = errorTextExisted.matchesSafely(textInputLayout)
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun testTextInputAction_ErrorTextExisted_describeToCorrect() {
+    val expectedErrorText = "Incorrect Administrator PIN. Please try again."
+    val expectedDescription =
+      "The expected error text is 'Incorrect Administrator PIN. Please try again.'"
+
+    val errorTextExisted = ErrorTextExisted(expectedErrorText)
+    var desc: Description = Description.NullDescription()
+    errorTextExisted.describeTo(desc)
+
+//    println("desc = $desc")
+//    println("expectedDescription = $expectedDescription")
+
+    assertThat(desc).isEqualTo(expectedDescription)
+  }
```

Comment:
@BenHenning I am having some problem writing tests for ```describeTo``` methods. I randomly tried doing this but this is not working as ```desc``` is always null with this. Any suggestion on how do I proceed to test describeTo methods?

Reply:
You may want to create a StringDescription instead: http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/StringDescription.html as the base. Does that work?

Reply:
Yes, Thanks